tests/test_modeling_fsmt.py

```diff
@@ -0,0 +1,509 @@
+
+        def _get_embs(m):
+            return (m.get_input_embeddings().weight.data.clone(), m.get_output_embeddings().weight.data.clone())
+
+        model = FSMTForConditionalGeneration(config).eval().to(torch_device)
+
+        # not equal in FSMT
+        # input, output = _get_embs(model)
+        # self.assertTrue(torch.eq(input, output).all(), msg=f"\n{input}\n{output}")
+
+        new_src_vocab_size = 45
+        model.resize_token_embeddings(new_src_vocab_size)
+        input_new, output_new = _get_embs(model)
+        self.assertEqual(
+            input_new.shape,
+            (new_src_vocab_size, config.d_model),
+            msg=f"input {input_new.shape}, {(new_src_vocab_size, config.d_model)}",
+        )
+        self.assertEqual(
+            output_new.shape,
+            (new_src_vocab_size, config.d_model),
+            msg=f"output {input_new.shape}, {(new_src_vocab_size, config.d_model)}",
+        )
+        self.assertTrue(torch.eq(input_new, output_new).all(), msg=f"{input_new}, {output_new}")
+
+
+def _assert_tensors_equal(a, b, atol=1e-12, prefix=""):
+    """If tensors not close, or a and b arent both tensors, raise a nice Assertion error."""
+    if a is None and b is None:
+        return True
+    try:
+        if torch.allclose(a, b, atol=atol):
+            return True
+        raise
+    except Exception:
+        msg = "{} != {}".format(a, b)
+        if prefix:
+            msg = prefix + ": " + msg
+        raise AssertionError(msg)
+
+
+def _long_tensor(tok_lst):
+    return torch.tensor(tok_lst, dtype=torch.long, device=torch_device)
+
+
+TOLERANCE = 1e-4
+
+
+@require_torch
+class FSMTModelIntegrationTests(unittest.TestCase):
+    @cached_property
+    def default_tokenizer(self):
+        return FSMTTokenizer.from_pretrained("stas/fsmt-wmt19-ru-en")
+
+    @slow
+    def test_inference_no_head(self):
+        tokenizer = self.default_tokenizer
+        model = FSMTModel.from_pretrained("stas/fsmt-wmt19-ru-en").to(torch_device)
+
+        src_text = "My friend computer will translate this for me"
+        input_ids = tokenizer([src_text], return_tensors="pt")["input_ids"]
+        input_ids = _long_tensor(input_ids)
+        inputs_dict = prepare_fsmt_inputs_dict(model.config, input_ids)
+        with torch.no_grad():
+            output = model(**inputs_dict)[0]
+        expected_shape = torch.Size((1, model.config.decoder_attention_heads, model.config.tgt_vocab_size))
+        self.assertEqual(output.shape, expected_shape)
+        expected_slice = torch.tensor(
+            [[-3.1850, -3.1849, 2.9694], [-4.0242, -4.0242, 0.2494], [-3.4442, -3.4443, 0.3315]], device=torch_device
+        )
+        print(output[:, :3, :3])
+        self.assertTrue(torch.allclose(output[:, :3, :3], expected_slice, atol=TOLERANCE))
+
+        # XXX: the rest of the tests were moved to tests/test_tokenization_bart.py - port from there into tokenization tests
+
+    @slow
+    def test_translation(self):
```

Comment:
(nit)

I would write three tests here so that it's easy to run one of them with 
```bash
RUN_SLOW=1 pytest tests/test_modeling_fsmt.py -k en-ru
```

It's also easy to see if all are broken or just 1.

Reply:
great idea. I will just wait if I can get `parameterized` available in our tests, as it'll be much easier to do:
https://github.com/huggingface/transformers/pull/6958